File: frontend/src/types/comments.ts
// src/types/comment.ts

export interface CommentBase {
  product_id: number;
  content: string;
  rating: number;
  created_at: string;       // datetime as ISO string
  sentiment_score: number;
  sentiment_label: string;
}

export interface CommentOut extends CommentBase {}

export interface CommentsOut {
  data: CommentBase[];
}

export interface CommentCreate {
  content: string;
  rating: number;
}

export interface CommentUpdate extends CommentCreate {}

File: app/schemas/comments.py
from pydantic import BaseModel
from datetime import datetime
from typing import List


class CommentBase(BaseModel):
    id: int
    product_id: int
    content: str
    rating: float
    created_at: datetime
    sentiment_score: float
    sentiment_label: str

    class Config:
        from_attributes = True


class CommentOut(CommentBase):
    pass


class CommentsOut(BaseModel):
    data: List[CommentBase]


class CommentCreate(BaseModel):
    content: str
    rating: float

    class Config:
        from_attributes = True


class CommentUpdate(CommentCreate):
    pass
